controllers/storagecluster/provider_server.go

```diff
@@ -185,27 +186,79 @@ func (o *ocsProviderServer) createService(r *StorageClusterReconciler, instance
 
 	r.Log.Info("Service create/update succeeded")
 
+	if instance.Spec.ProviderAPIServerServiceType == corev1.ServiceTypeLoadBalancer {
+		endpoint := o.getLoadBalancerEndpoint(actualService)
+
+		if endpoint == "" {
+			r.recorder.ReportIfNotPresent(instance, corev1.EventTypeNormal, "Waiting", "Waiting for Ingress on service "+actualService.Name)
+			r.Log.Info("Waiting for Ingress on service", "Service", actualService.Name, "Status", actualService.Status)
+			return reconcile.Result{RequeueAfter: time.Second * 5}, nil
+		}
+
+		instance.Status.StorageProviderEndpoint = fmt.Sprintf("%s:%d", endpoint, ocsProviderServicePort)
+	} else {
+		nodeAddresses, err := o.getWorkerNodesInternalIPAddresses(r)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+
+		if len(nodeAddresses) == 0 {
+			err = fmt.Errorf("Did not found any worker node addresses")
+			r.Log.Error(err, "Worker nodes count is zero")
+			return reconcile.Result{}, err
+		}
+
+		instance.Status.StorageProviderEndpoint = nodeAddresses[0] + ":" + fmt.Sprint(ocsProviderServiceNodePort)
```

Comment:
Can you pls use the same format as line no. 198

Reply:
done